File: app/src/main/java/piotr/celowski/shopperapp/presentation/shoppinglistdetails/ShoppingListDetailsFragment.java
package piotr.celowski.shopperapp.presentation.shoppinglistdetails;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import javax.inject.Inject;

import dagger.hilt.android.AndroidEntryPoint;
import piotr.celowski.shopperapp.domain.usecases.GroceryItemUseCases;

@AndroidEntryPoint
public class ShoppingListDetailsFragment extends Fragment {
    private int shoppingListId = 0;
    private Boolean shoppingListArchivedStatus = false;
    private ShoppingListDetailsView shoppingListDetailsView;

    @Inject GroceryItemUseCases groceryItemUseCases;
    @Inject ShoppingListDetailsController shoppingListDetailsController;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        shoppingListId = getArguments().getInt("shoppingListId");
        shoppingListArchivedStatus = getArguments().getBoolean("shoppingListArchived");

        shoppingListDetailsView = new ShoppingListDetailsView(
                inflater, container, shoppingListId, shoppingListArchivedStatus,
                getContext(), groceryItemUseCases, shoppingListDetailsController);
        shoppingListDetailsView.initFragment();
        shoppingListDetailsView.initRecyclerView();
        shoppingListDetailsView.setupOnClickListeners();

        shoppingListDetailsController.updateCacheWithDatabase();

        return shoppingListDetailsView.getRootView();
    }
}

File: app/src/main/java/piotr/celowski/shopperapp/presentation/shoppingListDetails/ShoppingListDetailsFragment.kt
package piotr.celowski.shopperapp.presentation.shoppingListDetails

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.view.isVisible
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import dagger.hilt.android.AndroidEntryPoint
import piotr.celowski.shopperapp.R
import piotr.celowski.shopperapp.databinding.FragmentShoppingListDetailsBinding
import piotr.celowski.shopperapp.domain.interfaces.ShoppingListWithGroceryItemsDAO
import piotr.celowski.shopperapp.domain.usecases.GroceryItemUseCases
import piotr.celowski.shopperapp.domain.usecases.ShoppingListUseCases
import piotr.celowski.shopperapp.presentation.shoppingLists.ShoppingListsController
import javax.inject.Inject

@AndroidEntryPoint
class ShoppingListDetailsFragment: Fragment() {
    private lateinit var mFragmentShoppingListDetailsBinding: FragmentShoppingListDetailsBinding
    private lateinit var mShoppingListDetailsRecycler: RecyclerView
    private var shoppingListId: Int = 0
    private var shoppingListArchivedStatus: Boolean = false

    @Inject
    lateinit var shoppingListsWithGroceryItemsDAO: ShoppingListWithGroceryItemsDAO

    @Inject
    lateinit var grocerItemUseCases: GroceryItemUseCases

    @Inject
    lateinit var shoppingListDetailsController: ShoppingListDetailsController

    override fun onCreateView(
            inflater: LayoutInflater, container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        mFragmentShoppingListDetailsBinding = FragmentShoppingListDetailsBinding.inflate(
                layoutInflater,
                container,
                false
        )

        shoppingListId = arguments?.getInt("shoppingListId")!!
        shoppingListArchivedStatus = arguments?.getBoolean("shoppingListArchived")!!

        val recyclerFragment = inflater.inflate(R.layout.fragment_shopping_list_details, container, false)

        mShoppingListDetailsRecycler = mFragmentShoppingListDetailsBinding.shoppingListDetailsRecyclerView
        mShoppingListDetailsRecycler.layoutManager = LinearLayoutManager(context)
        mShoppingListDetailsRecycler.adapter = ShoppingListDetailsAdapter(shoppingListId, shoppingListArchivedStatus, shoppingListsWithGroceryItemsDAO, grocerItemUseCases, shoppingListDetailsController)

        if(shoppingListArchivedStatus) {
            mFragmentShoppingListDetailsBinding.addGroceryItemFloatingButton.isVisible = false
        } else {
            mFragmentShoppingListDetailsBinding.addGroceryItemFloatingButton.setOnClickListener {
                shoppingListDetailsController.createNewGroceryItemForList(
                    "GroceryItem",
                    shoppingListId
                )
            }
        }

        shoppingListDetailsController.updateCacheWithDatabase()

        return mFragmentShoppingListDetailsBinding.root
    }
}